drivers/dma/dma_stm32.c

```diff
@@ -158,16 +174,32 @@ static int dma_stm32_get_priority(uint8_t priority, uint32_t *ll_priority)
 {
 	switch (priority) {
 	case 0x0:
+#if defined(CONFIG_DMA_STM32_V3)
+		*ll_priority = LL_DMA_LOW_PRIORITY_LOW_WEIGHT;
+#else
 		*ll_priority = LL_DMA_PRIORITY_LOW;
+#endif /* CONFIG_DMA_STM32_V3 */
 		break;
 	case 0x1:
+#if defined(CONFIG_DMA_STM32_V3)
+		*ll_priority = LL_DMA_LOW_PRIORITY_MID_WEIGHT;
+#else
 		*ll_priority = LL_DMA_PRIORITY_MEDIUM;
+#endif /* CONFIG_DMA_STM32_V3 */
 		break;
 	case 0x2:
+#if defined(CONFIG_DMA_STM32_V3)
+		*ll_priority = LL_DMA_LOW_PRIORITY_HIGH_WEIGHT;
+#else
 		*ll_priority = LL_DMA_PRIORITY_HIGH;
+#endif /* CONFIG_DMA_STM32_V3 */
 		break;
 	case 0x3:
+#if defined(CONFIG_DMA_STM32_V3)
+		*ll_priority = LL_DMA_HIGH_PRIORITY;
+#else
 		*ll_priority = LL_DMA_PRIORITY_VERYHIGH;
+#endif /* CONFIG_DMA_STM32_V3 */
```

Comment:
Can we use an array or something else to get rid of all these #ifdef's ?

Reply:
that could be a new table
```
static uint32_t table_priority[4] = {
#if defined(CONFIG_DMA_STM32_V3)
	LL_DMA_LOW_PRIORITY_LOW_WEIGHT,
	LL_DMA_LOW_PRIORITY_MID_WEIGHT,
	LL_DMA_LOW_PRIORITY_HIGH_WEIGHT,
	LL_DMA_HIGH_PRIORITY,
#else
	LL_DMA_PRIORITY_LOW,
	LL_DMA_PRIORITY_MEDIUM,
	LL_DMA_PRIORITY_HIGH,
	LL_DMA_PRIORITY_VERYHIGH,
#endif /* CONFIG_DMA_STM32_V3 */
};
```
used by the dma_stm32_get_priority() function as follows:

```
	if (priority > ARRAY_SIZE(table_priority)) {
		LOG_ERR("Priority error. %d", priority);
		return -EINVAL;
	}

	*ll_priority = table_priority[priority];
	return 0;
```